deepchecks/vision/vision_data/format_validators.py

```diff
@@ -0,0 +1,208 @@
+    name_plural = name + 's'
+    if task_type == TaskType.OTHER and data is not None:
+        raise ValidationError(f'The task type is OTHER, but {name_plural} were provided.')
+    try:
+        _ = data[0]
+        data = sequence_to_numpy(data)
+    except IndexError as err:
+        raise ValidationError(f'The batch {name_plural} must be a non empty iterable.') from err
+    except TypeError as err:
+        raise ValidationError(f'The batch {name_plural} must be an iterable, received {type(data)}') from err
+
+    sample_idx = 0
+    while data[sample_idx] is None or (isinstance(data[sample_idx], np.ndarray) and data[sample_idx].shape[0] == 0):
+        sample_idx += 1
+        if sample_idx == len(data):
+            return None  # No data to validate
+
+    if task_type == TaskType.CLASSIFICATION:
+        return data[sample_idx]  # for classification, the data is a single number no need to validate shape
+    try:
+        sample_shape = data[sample_idx].shape
+    except AttributeError as err:
+        raise ValidationError(f'{name} for {task_type.value} per image must be a multi dimensional array.') from err
+
+    if task_type == TaskType.OBJECT_DETECTION and len(sample_shape) not in (1, 2):
+        raise ValidationError(f'{name} for object detection per image must be a 2D array. Found shape {sample_shape}')
+
+    return data[sample_idx]
+
+
+def validate_additional_data_format(additional_data_batch):
+    """Validate that the data is in the required format.
+
+    Parameters
+    ----------
+    additional_data_batch
+        The additional data of the batch. Result of VisionData's batch_to_additional_data
+    Raises
+    ------
+    DeepchecksValueError
+        If the images doesn't fit the required deepchecks format.
+    """
+    if not hasattr(additional_data_batch, '__getitem__'):
+        raise ValidationError('The batch additional_data must be an iterable.')
+
+
+def validate_embeddings_format(embeddings):
+    """Validate that the data is in the required format.
+
+    Parameters
+    ----------
+    embeddings
+        The embeddings of the batch. Result of VisionData's batch_to_embeddings
+    Raises
+    ------
+    DeepchecksValueError
+        If the images doesn't fit the required deepchecks format.
+    """
+    if not hasattr(embeddings, '__getitem__'):
+        raise ValidationError('The batch embeddings must be an iterable.')
+
+
+def validate_image_identifiers_format(image_identifiers):
+    """Validate that the data is in the required format.
+
+    Parameters
+    ----------
+    image_identifiers
+        The image identifiers of the batch. Result of VisionData's batch_to_image_identifiers
+    Raises
+    ------
+    DeepchecksValueError
+        If the images doesn't fit the required deepchecks format.
+    """
+    try:
+        sample = image_identifiers[0]
+    except TypeError as err:
+        raise ValidationError('The batch image_identifiers must be an iterable.') from err
```

Comment:
sequence + maybe it is of len 0?
if image_identifiers is validated somewhere before for len validate it has getitem too

Reply:
only validated here, but currently has no functionality and format is not yet defined to details

Reply:
so fix validations and error description